includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -2354,4 +2375,48 @@ private function get_upe_gateway_id_for_order( $payment_method ) {
 
 		return $this->id;
 	}
+
+	/**
+	 * Fetches the appearance settings for Stripe Elements from the transient cache.
+	 *
+	 * @param bool $is_block_checkout Whether the appearance settings are for the block checkout.
+	 *
+	 * @return array|null The appearance settings.
+	 */
+	private function get_appearance_transient_key( $is_block_checkout = false ) {
+		return ( $is_block_checkout ? self::BLOCKS_APPEARANCE_TRANSIENT : self::APPEARANCE_TRANSIENT ) . '_' . get_option( 'stylesheet' );
```

Comment:
`get_option( 'stylesheet' )` would be the theme, right? like `storefront` for example... if that's the case, why a different transient per theme?

Reply:
> get_option( 'stylesheet' ) would be the theme, right?

Yeah, it's the theme name. I included it in the transient key because when you install and activate a new theme, our [`getAppearance()`](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/b67203690249e96cb7b41cac1413d0e71524c1ac/client/styles/upe/index.js#L312) JS function will generate a different set of rules depending on the theme.

WooPayments [clears their transients on `theme_switch`](https://github.com/Automattic/woocommerce-payments/blob/c6ebbaf8af2e95be8345f37f285518090f4495f7/includes/class-wc-payments-checkout.php#L101), but I decided to go with including the theme name in the transient key to be consistent with how we previously stored the localStorage against the current theme ([see here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/2e844deabc7dac8d73bc506393fc3104c1a78a16/client/stripe-utils/utils.js#L461-L462))